spec/lib/wos_queries_spec.rb

```diff
@@ -29,6 +31,25 @@
     end
   end
 
+  describe '#search_by_doi' do
+    let(:doi) { '10.1007/s12630-011-9462-1' }
+
+    it 'works' do
+      savon.expects(:authenticate).returns(wos_auth_response)
+      savon.expects(:search).with(message: :any).returns(wos_search_by_doi_response)
+      result = wos_queries.search_by_doi(doi)
+      expect(result).to be_an WosRecords
+      expect(result.count).to eq 1
+    end
+    it 'returns nothing for partial matches' do
+      savon.expects(:authenticate).returns(wos_auth_response)
+      savon.expects(:search).with(message: :any).returns(wos_search_by_doi_mismatch_response)
+      result = wos_queries.search_by_doi(doi)
+      expect(result).to be_an WosRecords
+      expect(result).to be_empty
```

Comment:
took me a second to understand why the same DOI passed in the two tests led to different results ... which of course works since the response is mocked differently as needed (i.e. the actual value of the incoming DOI is irrelevant for the tests) - maybe a small comment next to the let(:doi) statement to point that out?